scaleway/resource_baremetal_server.go

```diff
@@ -272,6 +309,58 @@ func resourceScalewayBaremetalServerUpdate(ctx context.Context, d *schema.Resour
 		return diag.FromErr(err)
 	}
 
+	server, err := baremetalAPI.GetServer(&baremetal.GetServerRequest{
+		Zone:     zonedID.Zone,
+		ServerID: zonedID.ID,
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var serverGetOptionIDs []string
+	for i := range server.Options {
+		serverGetOptionIDs = append(serverGetOptionIDs, server.Options[i].ID)
+	}
+
+	if d.HasChange("option_ids") {
+		optionIDs := expandStrings(d.Get("option_ids").(*schema.Set).List())
+		if len(optionIDs) > 0 {
+			optionsToAdd := baremetalCompareOptionIDsToAdd(optionIDs, serverGetOptionIDs, zonedID.Zone)
```

Comment:
Agree with @remyleone  we need a set(patch) here

Reply:
It has been asked to the product team, il will take time to be done, for now we'll do without it. I'll refacto the code when the setOption method is ready